src/network/netdev/bareudp.c

```diff
@@ -21,15 +21,11 @@ DEFINE_CONFIG_PARSE_ENUM(config_parse_bare_udp_iftype, bare_udp_protocol, BareUD
                          "Failed to parse EtherType=");
 
 static int netdev_bare_udp_fill_message_create(NetDev *netdev, Link *link, sd_netlink_message *m) {
-        BareUDP *u;
-        int r;
-
         assert(netdev);
         assert(m);
 
-        u = BAREUDP(netdev);
-
-        assert(u);
+        BareUDP *u = ASSERT_PTR(BAREUDP(netdev));
```

Comment:
That might be a good idea, but it's there's a huge number of those cases defined:
```
src/network/netdev/bareudp.h:DEFINE_NETDEV_CAST(BAREUDP, BareUDP);
src/network/netdev/batadv.h:DEFINE_NETDEV_CAST(BATADV, BatmanAdvanced);
src/network/netdev/bond.h:DEFINE_NETDEV_CAST(BOND, Bond);
src/network/netdev/bridge.h:DEFINE_NETDEV_CAST(BRIDGE, Bridge);
src/network/netdev/dummy.h:DEFINE_NETDEV_CAST(DUMMY, Dummy);
src/network/netdev/fou-tunnel.h:DEFINE_NETDEV_CAST(FOU, FouTunnel);
src/network/netdev/geneve.h:DEFINE_NETDEV_CAST(GENEVE, Geneve);
src/network/netdev/ifb.h:DEFINE_NETDEV_CAST(IFB, IntermediateFunctionalBlock);
src/network/netdev/ipoib.h:DEFINE_NETDEV_CAST(IPOIB, IPoIB);
src/network/netdev/ipvlan.h:DEFINE_NETDEV_CAST(IPVLAN, IPVlan);
src/network/netdev/ipvlan.h:DEFINE_NETDEV_CAST(IPVTAP, IPVlan);
src/network/netdev/l2tp-tunnel.h:DEFINE_NETDEV_CAST(L2TP, L2tpTunnel);
src/network/netdev/macsec.h:DEFINE_NETDEV_CAST(MACSEC, MACsec);
src/network/netdev/macvlan.h:DEFINE_NETDEV_CAST(MACVLAN, MacVlan);
src/network/netdev/macvlan.h:DEFINE_NETDEV_CAST(MACVTAP, MacVlan);
src/network/netdev/netdevsim.h:DEFINE_NETDEV_CAST(NETDEVSIM, NetDevSim);
src/network/netdev/nlmon.h:DEFINE_NETDEV_CAST(NLMON, NLMon);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(IPIP, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(GRE, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(GRETAP, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(IP6GRE, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(IP6GRETAP, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(SIT, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(VTI, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(VTI6, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(IP6TNL, Tunnel);
src/network/netdev/tunnel.h:DEFINE_NETDEV_CAST(ERSPAN, Tunnel);
src/network/netdev/tuntap.h:DEFINE_NETDEV_CAST(TUN, TunTap);
src/network/netdev/tuntap.h:DEFINE_NETDEV_CAST(TAP, TunTap);
src/network/netdev/vcan.h:DEFINE_NETDEV_CAST(VCAN, VCan);
src/network/netdev/veth.h:DEFINE_NETDEV_CAST(VETH, Veth);
src/network/netdev/vlan.h:DEFINE_NETDEV_CAST(VLAN, VLan);
src/network/netdev/vrf.h:DEFINE_NETDEV_CAST(VRF, Vrf);
src/network/netdev/vxcan.h:DEFINE_NETDEV_CAST(VXCAN, VxCan);
src/network/netdev/vxlan.h:DEFINE_NETDEV_CAST(VXLAN, VxLan);
src/network/netdev/wireguard.h:DEFINE_NETDEV_CAST(WIREGUARD, Wireguard);
src/network/netdev/wlan.h:DEFINE_NETDEV_CAST(WLAN, WLan);
src/network/netdev/xfrm.h:DEFINE_NETDEV_CAST(XFRM, Xfrm);
```
At least in some cases the code checks for NULL as a non-assert condition. So I think this can be done, but I'd prefer to do it separately.